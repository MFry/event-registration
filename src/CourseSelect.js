import React from "react";
import { withStyles } from "@material-ui/core/styles";
import FormControl from "@material-ui/core/FormControl";
import FormHelperText from "@material-ui/core/FormHelperText";
import InputLabel from "@material-ui/core/InputLabel";
import Select from "react-select";
import * as Animated from "react-select/lib/animated";
import PropTypes from "prop-types";

const styles = {
  select: {
    display: "inlineFlex",
    padding: 0,
    position: "relative"
  }
};

const CourseSelect = ({ courses }) => {
  return (
    <FormControl style={{ width: "100%" }}>
      <InputLabel htmlFor="course-selector">Course Select</InputLabel>
      <Select
        isMulti
        name="courses"
        placeholder="Select courses..."
        closeMenuOnSelect={false}
        components={Animated}
        options={courses}
        styles={styles.select}
      />
    </FormControl>
  );
};
CourseSelect.propTypes = {
  courses: PropTypes.array.isRequired
};
export default CourseSelect;
