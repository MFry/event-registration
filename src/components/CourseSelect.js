import React from "react";
import FormControl from "@material-ui/core/FormControl";
// Possible future addition
// import FormHelperText from "@material-ui/core/FormHelperText";
import InputLabel from "@material-ui/core/InputLabel";
import Select from "react-select";
import * as Animated from "react-select/lib/animated";
import PropTypes from "prop-types";

const styles = {
  select: {
    display: "inlineFlex",
    padding: 0,
    position: "relative"
  }
};

const CourseSelect = ({ courses }) => {
  return (
    <FormControl style={{ width: "100%" }}>
      <InputLabel htmlFor="course-selector">Course Select</InputLabel>
      <Select
        isMulti
        name="courses"
        placeholder="Select courses..."
        closeMenuOnSelect={false}
        autoBlur
        components={Animated}
        options={courses}
        styles={styles.select}
      />
    </FormControl>
  );
};
CourseSelect.propTypes = {
  courses: PropTypes.array.isRequired
};
export default CourseSelect;
