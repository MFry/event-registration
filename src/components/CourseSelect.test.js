import React from "react";
import ReactDOM from "react-dom";
import CourseSelect from "./CourseSelect";

it("renders without crashing", () => {
  const courses = [
    { value: "class1", label: "Class 1" },
    { value: "class2", label: "Class 2" },
    { value: "class3", label: "Class 3" },
    { value: "class4", label: "Class 4" }
  ];

  const div = document.createElement("div");
  ReactDOM.render(<CourseSelect courses={courses} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
