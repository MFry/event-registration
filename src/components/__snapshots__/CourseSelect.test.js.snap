// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div
  class="MuiFormControl-root-15"
  style="width:100%"
>
  <label
    class="MuiFormLabel-root-24 MuiInputLabel-root-19 MuiInputLabel-formControl-20 MuiInputLabel-animated-23"
    data-shrink="false"
    for="course-selector"
  >
    Course Select
  </label>
  <div
    class="css-10nd86i"
  >
    <span
      aria-atomic="true"
      aria-live="polite"
      class="css-1uvvha"
      role="status"
    >
      0 results available.
    </span>
    <div
      class="css-1aya2g8"
    >
      <div
        class="css-1rtrksz"
      >
        <div
          class="css-1492t68"
          style="opacity:1;transition:opacity 260ms"
        >
          Select courses...
        </div>
        <div
          class="css-rsyb7x"
        >
          <div
            class=""
            style="display:inline-block"
          >
            <input
              aria-autocomplete="list"
              aria-busy="false"
              aria-expanded="false"
              aria-haspopup="false"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              id="react-select-3-input"
              role="combobox"
              spellcheck="false"
              style="box-sizing:content-box;width:1px;background:0;border:0;font-size:inherit;opacity:1;outline:0;padding:0;color:inherit"
              tabindex="0"
              type="text"
              value=""
            >
              
            </input>
            <div
              style="position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre"
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1wy0on6"
      >
        <span
          class="css-d8oujb"
          role="presentation"
        >
          
        </span>
        <div
          class="css-1ep9fjw"
          role="button"
        >
          <svg
            class="css-19bqh2r"
            focusable="false"
            height="20"
            role="presentation"
            viewBox="0 0 20 20"
            width="20"
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            >
              
            </path>
          </svg>
        </div>
      </div>
    </div>
    <div>
      
    </div>
  </div>
</div>
`;
